umbraco/Umbraco-CMS
# V8: Validate uploaded files based on the configuration of the FileUpload property configuration

This fixes #10897.

Turns out that file extensions of uploaded files weren't validated at all based on the editor configuration, only the configuration in `umbracoSettings.config`. I've added extra validation so we now validate on both.

### Testing 

1. Create a document type with 2 `UploadField` properties, one that allows all extensions, and one that only accepts csv.
2. Try and create a content node, and ensure that:
   1. **Only** csv files are allowed in the limited upload field
   2. Other filetypes are allowed in the unlimited filetypes field
   3. Try adding your csv to the `disallowedUploadFiles` property of `umbracoSettings.config`, and ensure that you are no longer allowed to upload csv files, even to the upload field where you specified csv
   4. Try adding `<allowedUploadFiles>csv</allowedUploadFiles>` and ensure that only csv files are allowed, even in the upload field where everything is allowed
